src/PubSub/Subscription.php

```diff
@@ -0,0 +1,551 @@
+     *
+     * ```php
+     * <?php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $topic = $pubsub->topic('my-topic-name');
+     *
+     * $subscription = $topic->subscription('my-new-subscription');
+     * $result = $subscription->create();
+     * ```
+     * @param  array $options
+     * @return array An array of subscription info
+     */
+    public function create(array $options = [])
+    {
+        // If a subscription is created via PubSubClient::subscription(),
+        // it may or may not have a topic name. This is fine for most API
+        // interactions, but a topic name is required to create a subscription.
+        if (!$this->topicName) {
+            throw new \InvalidArgumentException('A topic name is required to
+                create a subscription.');
+        }
+
+        $this->info = $this->connection->createSubscription($options + [
+            'name' => $this->name,
+            'topic' => $this->topicName
+        ]);
+
+        return $this->info;
+    }
+
+    /**
+     * Delete a subscription
+     *
+     * ```php
+     * <?php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * $subscription->delete();
+     * ```
+     *
+     * @param  array $options
+     * @return void
+     */
+    public function delete(array $options = [])
+    {
+        $this->connection->deleteSubscription($options + [
+            'subscription' => $this->name
+        ]);
+    }
+
+    /**
+     * Check if a subscription exists.
+     *
+     * Service errors will NOT bubble up from this method. It will always return
+     * a boolean value. If you want to check for errors, use
+     * {@see Subscription::info()}.
+     *
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
+     *
+     * $cloud = new ServiceBuilder([
+     *      'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $pubsub = $cloud->pubsub();
+     *
+     * $subscription = $pubsub->subscription('my-new-subscription');
+     * if ($subscription->exists()) {
+     *      // do stuff
+     * }
+     * ```
+     *
+     * @param  array $options
+     * @return bool
+     */
+    public function exists(array $options = [])
+    {
+        try {
+            $this->info($options);
+            return true;
+        } catch (GoogleNotFoundException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * Get info on a subscription
+     *
+     * If the info is already cached on the object, it will return that result.
+     * To fetch a fresh result, use {@see Subscription::reload()}.
+     *
+     * ```php
+     * use Google\Cloud\ServiceBuilder;
```

Comment:
I'm noticing some examples start with `<?php` and others do not.  I think as a larger discussion we can consider which way to go on that, but for the sake of this PR can we just run with consistency and omit `<?php` for now?